modules/metrics_routing/variables.tf

```diff
@@ -13,10 +13,10 @@ variable "metrics_router_routes" {
   type = list(object({
     name = string
     rules = list(object({
-      action = string
-      targets = optional(list(object({
-        id = optional(string)
-      })), [])
+      action = optional(string, "send")
+      targets = list(object({
+        id = string
```

Comment:
Is this right?

targets are required for all actions, whilst the new validation implies that targets are only required for `send`, thus optional for any other action.

Reply:
The only reason why I have removed the optional from target is because in terraform registry it is required but at the same time the target is only required when the action is `send`. If a user wants to drop the metrics then they don't need a target. The example in the terraform [registry](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/metrics_router_route#example-of-route-with-drop-action) shows that.

Reply:
The other way I could think of is to keep the target as optional but remove the empty list as default and since there is validation then it makes sure that whenever the action is `send` the user has to input targets.

Reply:
I have made the target as optional & since there is validation for when the target is mandatory it makes sure the user inputs the target when required.